lib/heuristics/concerns/periodic_data_initialisation.rb

```diff
@@ -55,22 +55,22 @@ def initialize_routes(routes)
     routes.sort_by(&:day_index).each{ |defined_route|
       associated_route = @candidate_routes[defined_route.vehicle_id][defined_route.day_index.to_i]
       defined_route.mission_ids.each{ |id|
-        next if !@services_data.has_key?(id) # id has been removed when detecting unfeasible services in wrapper
+        next if !@services_data.key?(id) # id has been removed when detecting unfeasible services in wrapper
 
         best_index = find_best_index(id, associated_route, false) if associated_route
         considered_ids << id
         if best_index
-          insert_point_in_route(associated_route, best_index, false)
+          insert_point_in_route(associated_route, best_index, false, considered_ids.count(id))
 
           # unlock corresponding services
-          services_to_add = @services_unlocked_by[id].to_a - @uninserted.collect{ |_un, data| data[:original_id] }
+          services_to_add = @services_unlocked_by[id].to_a - @uninserted.collect{ |un_id, _v_nb, _r| un_id }
           @to_plan_service_ids += services_to_add
           @unlocked += services_to_add
         else
-          @uninserted["#{id}_#{considered_ids.count(id)}_#{@services_data[id][:raw].visits_number}"] = {
-            original_id: id,
-            reason: "Can not add this service to route (vehicle #{defined_route.vehicle_id}, day #{defined_route.day_index}) : already #{associated_route ? associated_route[:stops].size : 0} elements in route"
-          }
+          nb_elements = associated_route ? associated_route[:stops].size : 0
+          @uninserted << [id, considered_ids.count(id),
+                          "Can not add this service to route (vehicle #{defined_route.vehicle_id}," \
```

Comment:
```suggestion
                          "Can not add this service to route (vehicle #{defined_route.vehicle_id}, " \
```

Reply:
We need to end this string with a space or start the next one with it. otherwise the concatenation of the strings is incorrect